### PR TITLE
OpenTelemetry Agent support

### PR DESCRIPTION
**Pull request by pmm-sumo**
###### Description

When `otelagent.enabled=true`, deploys a DeamonSet with OpenTelemetry Collector running in Agent mode, configured to send data to the OpenTelemetry Collector instance.

###### Testing performed

- [ ] ci/build.sh
- [ ] Redeploy fluentd and fluentd-events pods
- [x] Confirm events, logs, and metrics are coming in

### CODE REVIEW

File: deploy/helm/sumologic/templates/otel-agent-configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "sumologic.metadata.name.otelagent.configmap" . }}
+  labels:
+    app: {{ template "sumologic.labels.app.otelagent.configmap" . }}
+    {{- include "sumologic.labels.common" . | nindent 4 }}
+data:
+  {{- (tpl (.Files.Glob "conf/traces/traces.otelagent.conf.yaml").AsConfig .) | nindent 2 }}
+{{- end }}
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/templates/otel-agent-configmap.yaml:11]:
```suggestion
{{- end }}

```

File: deploy/helm/sumologic/templates/otel-agent-daemonset.yaml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            port: 13133 # Health Check extension default port.
+        readinessProbe:
+          httpGet:
+            path: /
+            port: 13133 # Health Check extension default port.
+      volumes:
+        - configMap:
+            name: {{ template "sumologic.metadata.name.otelagent.configmap" . }}
+          name: otelagent-config-vol
+{{- end }}
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/templates/otel-agent-daemonset.yaml:98]:
```suggestion
{{- end }}

```

File: deploy/helm/sumologic/values.yaml [hunk 3]
```diff
@@ -1559,6 +1556,92 @@ falco:
   falco:
     jsonOutput: true
 
+
+##Configure optional OpenTelemetry Collector in Agent mode
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/values.yaml:1560]:
```suggestion
## Configure optional OpenTelemetry Collector in Agent mode
```

**Comment by sumo-drosiek** [comment]
@pmaciolek What is the purpose of the opentelemetry agent?

**Comment by pmm-sumo** [comment]
> @pmm-sumo What is the purpose of the opentelemetry agent?

Essentially, this makes scaling much more efficient. Agent gets the local connections, buffers data and sends those to collector. Eventually, tagging should move on there as well: https://opentelemetry.io/docs/collector/about/#agent

File: deploy/helm/sumologic/values.yaml [hunk 4]
```diff
@@ -163,9 +163,6 @@ sumologic:
   ## This is experimental feature and may be unavailable for your account
   traces:
     enabled: false
-    ## Use filter stdout plugin for tracing data
-    ## Warning: It will produce additional fluentd logs (one per trace)
-    fluentd_stdout: false
```
Comment (perk-sumo) [inline deploy/helm/sumologic/values.yaml:168]: Why this? Is it a breaking change?

**Review by perk-sumo — APPROVED** [review]
One nit, otherwise I'm 👍

File: deploy/helm/sumologic/values.yaml (hunk 4 above)
Comment (pmm-sumo) [reply deploy/helm/sumologic/values.yaml:168]: I believe this is a deprecated piece from when `fluentd` was taking care of traces. I didn't find any reference to it anywhere in the repo anymore

Comment (sumo-drosiek) [reply deploy/helm/sumologic/values.yaml:168]: correct 👍

File: deploy/helm/sumologic/values.yaml [hunk 5]
```diff
@@ -1559,6 +1556,92 @@ falco:
   falco:
     jsonOutput: true
 
+
+## Configure optional OpenTelemetry Collector in Agent mode
+otelagent:
+  enabled: false
+  daemonset:
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/values.yaml:1563]: please add
 - `extraEnvVars`
 - `extraVolumes`
 - `extraVolumeMounts`

**Review by sumo-drosiek — CHANGES_REQUESTED** [review]
Remove auto-generated files due to #1052

**Comment by pmm-sumo** [comment]
Sorry for the delay. The requests came when I was on a sick-leave and missed them. In either case - prepared the fixes. Going to test them bit later. But so far - does it look OK now @sumo-drosiek ?

**Comment by sumo-drosiek** [comment]
@pmm-sumo yup, this is fine. Just one another ask, please update README.md with the new configuration 😅

File: deploy/helm/sumologic/values.yaml [hunk 6]
```diff
@@ -1592,6 +1589,111 @@ falco:
...
+        memory: 196Mi
+        cpu: 50m
+    ## Add custom labels only to otelagent daemonset.
+    podLabels: {}
+    ## Add custom annotations only to otelagent daemonset.
+    podAnnotations: {}
+    ## Memory Ballast size should be max 1/3 to 1/2 of memory.
+    memBallastSizeMib: "250"
+    image:
+      name: "sumologic/opentelemetry-collector"
```
Comment (pmalek-sumo) [inline deploy/helm/sumologic/values.yaml:1613]:
```suggestion
      repository: "sumologic/opentelemetry-collector"
```

File: deploy/helm/sumologic/values.yaml [hunk 7]
```diff
@@ -1592,6 +1589,111 @@ falco:
...
+        insecure: true
+    service:
+      extensions: [health_check]
+      pipelines:
+        traces:
+          receivers: [jaeger, opencensus, otlp, zipkin]
+          processors: [memory_limiter, k8s_tagger, batch]
+          exporters: [otlp]
+
+
```
Comment (pmalek-sumo) [inline deploy/helm/sumologic/values.yaml:1696]:
```suggestion
          processors: [memory_limiter, k8s_tagger, batch]
          exporters: [otlp]
  
```

**Review by pmalek-sumo — COMMENTED** [review]
Other than those minor nitpicks I believe this is 👍🏻

**Review by pmalek-sumo — APPROVED** [review]

**Pull request merged**